app/assets/javascripts/save_as.js

```diff
@@ -1,40 +1,77 @@
-/*globals document,customPrompt*/
+/*globals CustomPrompt*/
 
-$(document).on('page:change', function () {
+var SaveAsForm = (function () {
     'use strict';
 
-    var nameInput = $("input#testing_ground_name");
-
     function submitSaveAsForm() {
-        var form = $(this).parents("form");
-        form.attr("action", $(this).data('url'));
-        form.submit();
+        this.form
+            .attr("action", this.formAction)
+            .submit();
     }
 
-    function promptCallback(newName) {
-        nameInput.val(newName);
+    function isValid(value) {
+        return (value.length < 100 && value.length > 0);
+    }
+
+    function callback(nameInput) {
+        var newName = nameInput.val(),
+            valid = isValid.call(this, newName);
+
+        nameInput.toggleClass("invalid", !valid)
+            .attr("title", function () {
+                return valid ? "" : "name is too long";
+            });
 
-        submitSaveAsForm.call(this);
+        if (valid) {
+            this.nameInput.val(newName);
+            this.customPrompt.popup.close();
+
+            submitSaveAsForm.call(this);
+        }
     }
 
     function onSaveAs(e) {
         e.preventDefault();
 
-        if (nameInput.hasClass('changed')) {
+        if (this.nameInput.hasClass('changed')) {
             submitSaveAsForm.call(this);
         } else {
-            customPrompt(
-                "Please set a name for your new LES",
-                nameInput.val(),
-                promptCallback.bind(this)
+            this.customPrompt = CustomPrompt.prompt(
+                "Please set a new name for your LES",
+                this.nameInput.val(),
+                callback.bind(this)
```

Comment:
I think it is possible to have `callback` take an explicit parameter. Instead of:

```js
function callback(nameInput) {
  // A wild "this" appears!
}
```

... you might try ...

```js
function callback(saveAsForm, nameInput) {
  // ...
}
```

... and then replace `callback.bind(this)` with `callback.bind(null, this)` (which creates a [partially-applied function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind#Partially_applied_functions)). The callback function will work identically when triggered by `CustomPrompt`, but has the benefit of being clearer about what argument its needs.